### PR TITLE
Dev sanjana

### PR DESCRIPTION
**Pull request by sanjanashah0110**

### CODE REVIEW

File: cmd/campaigns.go [hunk 1]
```diff
@@ -174,6 +174,15 @@ func handlePreviewCampaign(c echo.Context) error {
 	if id < 1 {
 		return echo.NewHTTPError(http.StatusBadRequest, app.i18n.T("globals.messages.invalidID"))
 	}
+	initSettings("SELECT JSON_OBJECT_AGG(key, value) AS settings FROM settings WHERE authid = $1;", db, ko, authID)
+	app.manager = initCampaignManager(app.queries, app.constants, app)
```
Comment (RaviChauhanSiprtc) [inline cmd/campaigns.go]: make a comman function for setting get

File: cmd/campaigns.go [hunk 2]
```diff
@@ -787,3 +854,31 @@ func makeOptinCampaignMessage(o campaignReq, app *App) (campaignReq, error) {
 	o.Body = b.String()
 	return o, nil
 }
+
+// GetCampaignReport retrieves the campaign reports.
+// If IDs are provided then, those specific campaign reports are returned, otherwise all campaign reports are returned.
+func handleGetCampaignsReport(c echo.Context) error {
```
Comment (RaviChauhanSiprtc) [inline cmd/campaigns.go:846]: put a logger.debug

File: cmd/handlers.go [hunk 3]
```diff
@@ -146,6 +146,7 @@ func initHTTPHandlers(e *echo.Echo, app *App) {
 	g.PUT("/api/campaigns/:id/status", handleUpdateCampaignStatus)
 	g.PUT("/api/campaigns/:id/archive", handleUpdateCampaignArchive)
 	g.DELETE("/api/campaigns/:id", handleDeleteCampaign)
+	g.GET("/api/campaigns-report", handleGetCampaignsReport)
```
Comment (RaviChauhanSiprtc) [inline cmd/handlers.go]: /api/campaigns-report

change to 

/api/campaigns/report

File: cmd/init.go [hunk 4]
```diff
@@ -600,28 +600,54 @@ func initPostbackMessengers(m *manager.Manager) []manager.Messenger {
 	return out
 }
 
+func parseS3Opts(data map[string]interface{}) s3.Opt {
```
Comment (RaviChauhanSiprtc) [inline cmd/init.go]: move to utils

File: cmd/init.go [hunk 5]
```diff
@@ -600,28 +600,54 @@ func initPostbackMessengers(m *manager.Manager) []manager.Messenger {
...
+		AccessKey:  data["aws_access_key_id"].(string),
+		SecretKey:  data["aws_secret_access_key"].(string),
+		Region:     data["aws_default_region"].(string),
+		Bucket:     data["bucket"].(string),
+		BucketPath: data["bucket_path"].(string),
+		BucketType: data["bucket_type"].(string),
+		Expiry:     expiry,
+	}
+}
+func parseFileSystemOpts(data map[string]interface{}) filesystem.Opts {
```
Comment (RaviChauhanSiprtc) [inline cmd/init.go]: move to utils

File: cmd/main.go [hunk 6]
```diff
@@ -154,9 +154,9 @@ func init() {
 	qMap := readQueries(queryFilePath, db, fs)
 
 	// Load settings from DB.
-	if q, ok := qMap["get-settings"]; ok {
-		initSettings(q.Query, db, ko)
-	}
+	// if q, ok := qMap["get-settings"]; ok {
+	// 	initSettings(q.Query, db, ko)
+	// }
```
Comment (RaviChauhanSiprtc) [inline cmd/main.go]: remove this

File: cmd/main.go [hunk 7]
```diff
@@ -169,7 +169,7 @@ func main() {
 		fs:         fs,
 		db:         db,
 		constants:  initConstants(),
-		media:      initMediaStore(),
+		// media:      initMediaStore(),
```
Comment (RaviChauhanSiprtc) [inline cmd/main.go]: also remove this

File: cmd/main.go [hunk 8]
```diff
@@ -208,7 +208,7 @@ func main() {
 	})
 
 	app.queries = queries
-	app.manager = initCampaignManager(app.queries, app.constants, app)
+	// app.manager = initCampaignManager(app.queries, app.constants, app)
```
Comment (RaviChauhanSiprtc) [inline cmd/main.go]: also remove this

**Review by RaviChauhanSiprtc — CHANGES_REQUESTED** [review]
make a common function

File: cmd/handlers.go (hunk 3 above)
Comment (sanjanashah0110) [reply cmd/handlers.go]: Done

File: cmd/main.go (hunk 6 above)
Comment (sanjanashah0110) [reply cmd/main.go]: Done

File: cmd/main.go (hunk 7 above)
Comment (sanjanashah0110) [reply cmd/main.go]: Done

**Pull request merged**